apps/google/src/inngest/functions/data-protection/refresh-object.ts

```diff
@@ -0,0 +1,116 @@
+  'google/data_protection.refresh_object.requested': RefreshDataProtectionObjectRequested;
+};
+
+type RefreshDataProtectionObjectRequested = {
+  data: {
+    organisationId: string;
+    objectId: string;
+    ownerId: string;
+  };
+};
+
+export const refreshDataProtectionObject = inngest.createFunction(
+  {
+    id: 'google-refresh-data-protection-object',
+    retries: 3,
+    concurrency: {
+      limit: 1,
+    },
+    cancelOn: [
+      {
+        event: 'google/common.remove_organisation.requested',
+        match: 'data.organisationId',
+      },
+    ],
+  },
+  { event: 'google/data_protection.refresh_object.requested' },
+  async ({
+    event: {
+      data: { organisationId, objectId, ownerId },
+    },
+    step,
+  }) => {
+    const user = await step.run('get-user', async () => {
+      return db.query.usersTable.findFirst({
+        where: and(eq(usersTable.organisationId, organisationId), eq(usersTable.id, ownerId)),
+        columns: {
+          email: true,
+        },
+        with: {
+          organisation: {
+            columns: {
+              region: true,
+            },
+          },
+        },
+      });
+    });
+
+    if (!user) {
+      throw new NonRetriableError('User not found');
+    }
+
+    const authClient = await getGoogleServiceAccountClient(user.email);
+    const elba = getElbaClient({ organisationId, region: user.organisation.region });
+
+    try {
+      const file = await step.run('get-file', async () => {
+        return getGoogleFile({
+          auth: authClient,
+          fileId: objectId,
+        });
+      });
+      let permissions: GoogleFileNonInheritedPermission[] = [];
+      let pageToken: string | undefined;
+      do {
+        // eslint-disable-next-line no-await-in-loop -- we need to await to get next page
+        const { permissions: googlePermissions, nextPageToken } = await step.run(
+          `get-permissions-${pageToken}`,
+          // eslint-disable-next-line @typescript-eslint/no-loop-func -- we iterate through pageToken
+          async () => {
+            return listGoogleFileNonInheritedPermissions({
+              auth: authClient,
+              fileId: objectId,
+              pageToken,
+              pageSize: 100,
+            });
+          }
+        );
+
+        permissions = [...permissions, ...googlePermissions];
+        pageToken = nextPageToken ?? undefined;
+      } while (pageToken);
+
+      const object = formatDataProtectionObject({ file, owner: ownerId, permissions });
+
+      await elba.dataProtection.updateObjects({ objects: [object] });
+
+      return { status: 'updated' };
+
+      /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access -- Start of error handling */
+    } catch (error: any) {
+      if (error?.code === 404) {
+        /* eslint-enable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access -- End of error handling */
```

Comment:
GAxiosError comes from `Common`, the other class specified earlier can also be used (it's extending from GAxiosError)
```suggestion
    } catch (error) {
      if (error instanceof GAxiosError && error.code === 404) {
```